### DISCUSSION TITLE
Which package manager to prefer and lock-in?

### DISCUSSION THREAD

**Discussion by swernerx** [vercel/turborepo#685] [Help] [answered]
Hi! We have planned to build our new foundation with dozens of subpackages using Turborepo.

At the beginning, I am directly wondering if I should rather create a new TurboRepo project based on Yarn or NPM. Is there a preference from the tooling side? Earlier, in Lerna, it was advantageous to use Yarn (because of the workspaces). What is the situation with Turborepo? Is there still a preference? What about the upcoming features in Yarn 2/3 - will this change the situation? 

Is there some kind of "lock-in" if you select the wrong package manager, or can you still change package managers later?

**Comment by mfpopa** [comment, marked as answer]
Turborepo supports NPM, Yarn and PNPM. There is nothing in the official documentation suggesting package manager lock-in, or anything special happening when choosing one over the others.

NPM v7 introduced workspaces, so now you can actually swap between NPM and Yarn whenever you want, both in Lerna and Turborepo.

**Comment by Cypher1** [comment]
Small follow up question, I'm trying to use pnpm or yarn and I can't get turbo to stop detecting npm.
How do I stop turbo from erroring on having multiple package managers?

**Reply by Cypher1** [reply]
Ah, Got it! It was the package-lock.json file. It was gitignored so I hadn't noticed that it was still around.
Thanks for the sweet tool!
